src/debug/debug.cc

```diff
@@ -4261,6 +4261,17 @@ GetOrCreatePromiseDependencyGraphData(Isolate* isolate, Handle<Object> promise)
   return iter->second;
 }
 
+void AddPromiseDependencyGraphAdoption(Isolate* isolate, Handle<Object> promise, Handle<Object> adopted) {
+  PromiseDependencyGraphData& data = GetOrCreatePromiseDependencyGraphData(isolate, promise);
+  PromiseDependencyGraphData& adopted_data = GetOrCreatePromiseDependencyGraphData(isolate, adopted);
```

Comment:
I think slightly, dilutes what `GetOrCreatePromiseDependencyGraphData` is about. By reading this code I'd think it would contain only promise IDs but that's not quite true as other objects could potentially be "registered" in it, right?

I also wonder if this whole thing can now generate a sensible graph when thenables are involved

Reply:
I agree with these concerns, GetOrCreatePromiseDependencyGraphData should only be used with promises.  Things won't break if it's used with other objects, but the edges being created here seem fairly meaningless.  I'd like a more detailed explanation of how this PR helps to be able to review it.